addons/point_of_sale/models/pos_config.py

```diff
@@ -320,19 +320,30 @@ def _compute_selectable_categories(self):
 
     @api.constrains('cash_control')
     def _check_session_state(self):
-        open_session = self.env['pos.session'].search([('config_id', '=', self.id), ('state', '!=', 'closed')])
+        open_session = self.env['pos.session'].search([('config_id', 'in', self.ids), ('state', '!=', 'closed')], limit=1)
         if open_session:
             raise ValidationError(_("You are not allowed to change the cash control status while a session is already opened."))
```

Comment:
This kind of constraint should ideally be handled in write call, because it only restrains allowed modifications, not real record state...
Furthermore, it will be triggered even if the record cash_control isn't really modified... 
`config.cash_control = config.cash_control`